x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardTaskParams.java

```diff
@@ -71,7 +71,9 @@ public record DownsampleShardTaskParams(
             new ShardId(in),
             in.readStringArray(),
             in.readStringArray(),
-            in.readOptionalStringArray()
+            in.getTransportVersion().onOrAfter(TransportVersions.DIMENSIONS_SERIALIZATION_PERSISTENT_TASK)
+                ? in.readOptionalStringArray()
+                : new String[] {}
```

Comment:
What happens when a task with no dimensions gets executed? Shall we update the executor to abort?

Reply:
The task is expected to fail with an unretryable error because of no dimensions.

Reply:
I think we should not fail, but use the old way of downsampling for tasks without dimensions, but I might be missing something obvious here.